src/terminal/terminal/terminal.h

```diff
@@ -279,6 +279,34 @@ guac_terminal_options* guac_terminal_options_create(
  */
 void guac_terminal_free(guac_terminal* term);
 
+/**
+ * Calculate the available space within the terminal 
+ * and store the results in the pointer arguments.
+ *
+ * @param terminal
+ *     The terminal provides character width and height for caluclations.
+ * 
+ * @param width
+ *     The width of the terminal, in pixels.
+ *
+ * @param height
+ *     The height of the terminal, in pixels.
+ * 
+ * @param aw
+ *     Set with the available width for text of the terminal, in pixels.
+ * 
+ * @param ah
+ *     Set with the available height for text of the terminal, in pixels.
+ * 
+ * @param columns
+ *     Set with the available width for text of the terminal, by column count.
+ * 
+ * @param rows
+ *     Set with the available height for text of the terminal, by row count.
+ */
+void get_available_dimensions(guac_terminal* term, int width, int height, 
+        int* aw, int* ah, int* columns, int* rows);
+
```

Comment:
Looks like this is only used in `terminal.c` - is there a reason to make it available in the header, or should it be a `static` function limited to `terminal.c`?